src/sentry/static/sentry/app/views/performance/transactionSummary/filter.tsx

```diff
@@ -0,0 +1,264 @@
+                return (
+                  <ListItem key={String(index)} isChecked={false}>
+                    <OperationDot backgroundColor={pickSpanBarColour(operationName)} />
+                    <OperationName>{operationName}</OperationName>
+                    <Radio
+                      radioSize="small"
+                      checked={filterOption === currentFilter}
+                      onChange={() => onChangeFilter(filterOption)}
+                    />
+                  </ListItem>
+                );
+              })}
+            </List>
+          </MenuContent>
+        </DropdownControl>
+      </Wrapper>
+    );
+  }
+}
+
+const FilterLabel = styled('span')`
+  margin-left: ${space(1)};
+`;
+
+const Wrapper = styled('div')`
+  position: relative;
+  display: flex;
+
+  margin-right: ${space(1)};
+`;
+
+const StyledDropdownButton = styled(DropdownButton)<{hasDarkBorderBottomColor?: boolean}>`
+  white-space: nowrap;
+  max-width: 200px;
+
+  z-index: ${p => p.theme.zIndex.dropdown};
+
+  &:hover,
+  &:active {
+    ${p =>
+      !p.isOpen &&
+      p.hasDarkBorderBottomColor &&
+      `
+          border-bottom-color: ${p.theme.button.primary.border};
+        `}
+  }
+
+  ${p =>
+    !p.isOpen &&
+    p.hasDarkBorderBottomColor &&
+    `
+      border-bottom-color: ${p.theme.button.primary.border};
+    `}
+`;
+
+const MenuContent = styled('div')`
+  max-height: 250px;
+  overflow-y: auto;
+  border-top: 1px solid ${p => p.theme.gray200};
+`;
+
+const Header = styled('div')`
+  display: grid;
+  grid-template-columns: auto min-content;
+  grid-column-gap: ${space(1)};
+  align-items: center;
+
+  margin: 0;
+  background-color: ${p => p.theme.backgroundSecondary};
+  color: ${p => p.theme.gray300};
+  font-weight: normal;
+  font-size: ${p => p.theme.fontSizeMedium};
+  padding: ${space(1)} ${space(2)};
+  border-bottom: 1px solid ${p => p.theme.border};
+`;
+
+const HeaderTitle = styled('span')`
+  font-size: ${p => p.theme.fontSizeMedium};
+`;
+
+const List = styled('ul')`
+  list-style: none;
+  margin: 0;
+  padding: 0;
+`;
+
+const ListItem = styled('li')<{isChecked?: boolean}>`
+  display: grid;
+  grid-template-columns: max-content 1fr max-content;
+  grid-column-gap: ${space(1)};
+  align-items: center;
+  padding: ${space(1)} ${space(2)};
+  border-bottom: 1px solid ${p => p.theme.border};
+  :hover {
+    background-color: ${p => p.theme.backgroundSecondary};
+  }
+
+  &:hover span {
+    color: ${p => p.theme.blue300};
+    text-decoration: underline;
+  }
+`;
+
+const OperationDot = styled('div')<{backgroundColor: string}>`
+  content: '';
+  display: block;
+  width: 8px;
+  min-width: 8px;
+  height: 8px;
+  margin-right: ${space(1)};
+  border-radius: 100%;
+
+  background-color: ${p => p.backgroundColor};
+`;
+
+const OperationName = styled('div')`
+  font-size: ${p => p.theme.fontSizeMedium};
+  ${overflowEllipsis};
+`;
+
+export function filterToString(option: SpanOperationBreakdownFilter): string {
```

Comment:
If you want to avoid this being a long converting function, you can either set enum values to these strings, or use a map to do a lookup.